drivers/modem/gsm_ppp.c

```diff
@@ -383,19 +389,34 @@ static void gsm_finalize_connection(struct gsm_modem *gsm)
 	}
 
 	/* Don't initialize PPP until we're attached to packet service */
-	ret = modem_cmd_send_nolock(&gsm->context.iface,
-				    &gsm->context.cmd_handler,
-				    &check_attached_cmd, 1,
-				    "AT+CGATT?",
-				    &gsm->sem_response,
-				    GSM_CMD_SETUP_TIMEOUT);
+	ret = modem_cmd_send_nolock(
+		&gsm->context.iface, &gsm->context.cmd_handler,
+		&check_attached_cmd, 1, "AT+CGATT?", &gsm->sem_response,
+		GSM_CMD_SETUP_TIMEOUT);
 	if (ret < 0) {
+		/*
+		 * attach_retry not set        -> trigger 10 attach retries
+		 * attach_retry set            -> decrement and retry
+		 * attach_retry set, becomes 0 -> trigger full retry
+		 *
+		 * In total, modem gets 60 seconds to ack AT+CGATT which should
+		 * be fine for most RATs but possibly a making num retries
+		 * Kconfigurable would be best eventually.
+		 */
+		if (!attach_retry) {
+			attach_retry = 10;
+		} else {
+			attach_retry--;
+		}
+
 		LOG_DBG("Not attached, %s", "retrying...");
 		(void)k_delayed_work_submit(&gsm->gsm_configure_work,
-					    K_SECONDS(1));
+					    K_SECONDS(3));
```

Comment:
Why do you increase that?

Reply:
Not sure, this patch has been on the shelf for a little while now. I believe it makes more sense with the latest force push.